Makefile

```diff
@@ -143,13 +144,7 @@ e2e-docker: metering-src-docker-build
 	docker cp metering-e2e-docker:/out bin/e2e-docker-test-output
 	docker rm metering-e2e-docker
 
-vet:
-	go vet $(GO_PKG)/cmd/... $(GO_PKG)/pkg/...
-
-verify: verify-codegen verify-olm-manifests verify-helm-templates fmt vet
-	@echo Checking for unstaged changes
-	# validates no unstaged changes exist in $(VERIFY_FILE_PATHS)
-	git diff --stat HEAD --ignore-submodules --exit-code -- $(VERIFY_FILE_PATHS)
```

Comment:
Hmm, I still think we want something similar to what `git diff` is doing here, even if it's only for validating go module changes, and more specifically checking if there are any changes made to go.mod, go.sum, vendor/ that are unstaged:
```make
...
    @git diff --exit-code -- go.sum go.mod vendor/
```

Reply:
I think I am not getting something? How would there come to be module changes, or changes made to go.mod, go.sum or vendor that are unstaged? Like by running `go build` or `go mod tidy` as part of the CI process? I guess those commands could cause there to be diffs but why do they matter? The committed code will produce those changes when run through the CI process, why worry about essentially build artifacts?

Reply:
It's essentially a stop-gap to ensure that changes made to dependencies _get_ staged. In the case where you're adding a new dependency and you follow the intended workflow, i.e. you run `make vendor` and then commit those changes, then there's no issue as there's going to be no unstaged changes.

We do the same thing with `make metering-manifests`: if you make a change to a helm chart, we expect you to run that command so that the corresponding manifests also get updated. If you don't follow that workflow, then verify will complain about it.

Reply:
Okay, I put it back in 9c6beb3 Pretty sure it will cause a fail, but we can look at how to avoid the fails after I guess.

Reply:
used Docker to run make update-codegen and commited results